File: service/app/src/main/java/io/miragon/blueprint/adapter/process/HistoryCleanupConfiguration.java
package io.miragon.blueprint.adapter.process;

import org.operaton.bpm.engine.ProcessEngine;
import org.operaton.bpm.engine.impl.cfg.ProcessEngineConfigurationImpl;
import org.operaton.bpm.engine.impl.cfg.ProcessEnginePlugin;
import org.springframework.context.annotation.Configuration;

/**
 * Process-engine plugin that configures removal-time-based history cleanup. Operaton auto-detects
 * beans implementing {@link ProcessEnginePlugin} and registers them with the embedded engine.
 */
@Configuration
public class HistoryCleanupConfiguration implements ProcessEnginePlugin {

    @Override
    public void preInit(ProcessEngineConfigurationImpl configuration) {
        configuration.setHistoryCleanupStrategy("removalTimeBased");
        configuration.setHistoryCleanupBatchWindowStartTime("22:00");
        configuration.setHistoryCleanupBatchWindowEndTime("06:00");
        configuration.setHistoryCleanupBatchSize(500);
        configuration.setHistoryCleanupDegreeOfParallelism(1);
    }

    @Override
    public void postInit(ProcessEngineConfigurationImpl configuration) {
    }

    @Override
    public void postProcessEngineBuild(ProcessEngine processEngine) {
    }
}

File: service/app/src/main/kotlin/io/miragon/blueprint/adapter/process/HistoryCleanupConfiguration.kt
package io.miragon.blueprint.adapter.process

import org.operaton.bpm.engine.ProcessEngine
import org.operaton.bpm.engine.impl.cfg.ProcessEngineConfigurationImpl
import org.operaton.bpm.engine.impl.cfg.ProcessEnginePlugin
import org.springframework.context.annotation.Configuration

/**
 * Process-engine plugin that configures removal-time-based history cleanup. Operaton auto-detects
 * beans implementing [ProcessEnginePlugin] and registers them with the embedded engine.
 */
@Configuration
class HistoryCleanupConfiguration : ProcessEnginePlugin {

    override fun preInit(configuration: ProcessEngineConfigurationImpl) {
        configuration.historyCleanupStrategy = "removalTimeBased"
        configuration.historyCleanupBatchWindowStartTime = "22:00"
        configuration.historyCleanupBatchWindowEndTime = "06:00"
        configuration.historyCleanupBatchSize = 500
        configuration.historyCleanupDegreeOfParallelism = 1
    }

    override fun postInit(configuration: ProcessEngineConfigurationImpl) {}

    override fun postProcessEngineBuild(processEngine: ProcessEngine) {}
}
